### PR TITLE
[CCI][BUG] Test support required for Python v3.9+

### LINKED ISSUE #323: [CCI][BUG] Test support required for Python v3.9+
### What is the bug?
When running test suite tests don't run for python 3.9+

### How can one reproduce the bug?
1. Install Python 3.9+
2. Run the `nox -rs test` command

### What is the expected behavior?
If you don't have the version specified in noxfile.py, all tests will be skipped:

nox > * test-2.7: failed
nox > * test-3.4: skipped
nox > * test-3.5: skipped
nox > * test-3.6: skipped
nox > * test-3.7: skipped
nox > * test-3.8: skipped
nox > * test-3.9: skipped

### What is your host/environment?
OS: macOS
Python: v3.10.4

### PR DESCRIPTION
**Pull request by bl1nkker**
### Description
Python 3.9+ support has been added

### Issues Resolved
#323 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Thanks, needs `-s` for DCO please.

**Comment by bl1nkker** [comment]
done 👍

**Comment by bl1nkker** [comment]
When I was working on another problem, I noticed this too. It was rather strange to see that the same lines of code for tests were repeated twice

In any case, those versions specified in ci.yml are required for testing before deployment. While the versions that are in noxfile are needed for testing locally. In fact, this tests are only needed for development purposes.

**Comment by dblock** [comment]
@bl1nkker Should we add a comment here to also update .yml for CI and vice-versa?

**Comment by bl1nkker** [comment]
done👍

also changed changelog.md

**Comment by bl1nkker** [comment]
@dblock Should i resolve this conflict? Or is it better to wait for a mentor?

**Comment by dblock** [comment]
You should always rebase/resolve conflicts in your PRs, and iterate to green until they are merged.

**Comment by bl1nkker** [comment]
yes, sorry for the delayed reply

**Comment by bl1nkker** [comment]
any idea why python 3.4-3.6 tests don't work?
image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,7 +38,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["2.7", "3.7", "3.8", "3.9", "3.10", "3.11"]
+        # Python versions used for testing. Any changes made here should also be reflected in the noxfile.py file.
+        python-version:
+          ["2.7", "3.4", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (VachaShah) [inline .github/workflows/ci.yml:43]: Version matrix can be kept the same. Python versions 3.5 and 3.6 need a specific os and hence are already included in line 47 and line 50 of this file.

**Comment by VachaShah** [comment]
> any idea why python 3.4-3.6 tests don't work? image

Please check my latest comment on why the tests are failing.

**Comment by saimedhi** [comment]
Hello, @bl1nkker. Please do not modify existing Python versions. It is sufficient to add versions 3.10 and 3.11 to the Nox file and the Github workflow. If you are unable to resolve merge conflicts, please close this PR and create a new PR with only the changes indicated. Thanks :)

**Comment by saimedhi** [comment]
Hello @bl1nkker, please complete the pull request.

**Pull request closed without merging**